src/fu-engine.c

```diff
@@ -882,10 +883,10 @@ fu_engine_vendor_quirk_release_version (AsApp *app)
 	if (as_app_get_kind (app) != AS_APP_KIND_FIRMWARE)
 		return;
 
-	for (guint i = 0; quirk_table[i].identifier != NULL; i++) {
-		if (g_str_has_prefix (as_app_get_id(app), quirk_table[i].identifier))
-			flags = quirk_table[i].flags;
-	}
+	/* FIXME: special case DELL, perhaps this can go in the XML?
+	 * com.dell.uefi*.firmware -> AS_VERSION_PARSE_FLAG_NONE */
+	if (g_str_has_prefix (as_app_get_id (app), "com.dell.uefi"))
+		flags = AS_VERSION_PARSE_FLAG_NONE;
 
```

Comment:
When you say maybe this can go in XML, do you mean LVFS output metadata or individual XML for uploads?

I think it's better to keep this directly in fwupd and let the engine parse the quirks like plugins do.  Although it's only being done for UEFI plugin I don't think this will necessarily only be a UEFI thing in the future.  Everyone can do their own interpretations of versions.

Reply:
I've added this now.